skia/RSAsyncStorageModule.cpp

```diff
@@ -28,49 +71,162 @@ auto RSAsyncStorageModule::getMethods() -> std::vector<Method> {
       Method(
           "mergeItem",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            mergeItem(args, cb);
           }),
       Method(
           "getAllKeys",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            getAllKeys(args, cb);
           }),
       Method(
           "clear",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            clear(args, cb);
           }),
 
   };
 }
 
+
+void RSAsyncStorageModule::multiGet(dynamic args, CxxModule::Callback cb) {
+  taskRunner_->dispatch( [this,args,cb](){
+    dynamic errors = folly::dynamic::array;
+    dynamic resultArray = folly::dynamic::array;
+    for (auto& itemArray : args[0]) { 
+      auto pos = appLocalDataFile_.find(itemArray);
+      dynamic errorItem = folly::dynamic::object();
+      if(pos == appLocalDataFile_.items().end()){
+        errorItem["message"] = "Failed to found the get Item";
+        errorItem["key"] = itemArray;
+        errors.push_back(errorItem);
+      }else{
+        dynamic result = folly::dynamic::array;
+        result.push_back(itemArray);
+        result.push_back(appLocalDataFile_[itemArray]);
+        resultArray.push_back(result);
+      }
+    }
+    cb({(errors.empty() ? nullptr : errors), resultArray});
+  });  
+}
+
+void RSAsyncStorageModule::multiSet(dynamic args, CxxModule::Callback cb) {
+  if(!appLocalFile_.is_open()) {
+    dynamic errors = folly::dynamic::object();
+    errors["message"] = "Failed to write the data";
+    errors["key"] = args[0][0].asString();
+    cb({folly::dynamic::array(errors)});
+    return;
+  }
+  taskRunner_->dispatch( [this,args,cb]() {
+    int requiredLength = 0;
+    for (auto& itemArray : args[0]) {
+      int oldLength = 0;
+      int newLength = (itemArray[0].asString()).length()+itemArray[1].asString().length()+6;
+      // value size cannot be larger than 2 MB
+      if((itemArray[1].asString()).length() < ASYNC_VALUE_DEFAULT_MAX_CACHE_LIMIT ) {
+        if((appLocalDataFile_.find(itemArray[0].asString())) != appLocalDataFile_.items().end()) {
+          oldLength = (itemArray[0].asString()).length()+(appLocalDataFile_[itemArray[0]].asString()).length()+6;
+          if(oldLength < newLength) {
+            requiredLength = newLength - oldLength;
+          }else{
+            requiredLength = oldLength-newLength;
+          }
+        }else{
+          requiredLength= newLength;
+        }
+        appLocalDataFile_[itemArray[0].asString()] = itemArray[1].asString();
+      }
+    }
+    totalSize_ += requiredLength;
+    if(isWriteScheduled_)
+      return;
```

Comment:
All these should be done for every key. Should be inside for loop.
Need to schedule only one file write after for loop ( you can use a local bool flag )
set global flag once we schedule